br/pkg/aws/ebs.go

```diff
@@ -251,6 +251,9 @@ func (e *EC2Session) WaitSnapshotsCreated(snapIDMap map[string]string, progress
 			if *s.State == ec2.SnapshotStateCompleted {
 				log.Info("snapshot completed", zap.String("id", *s.SnapshotId))
 				totalVolumeSize += *s.VolumeSize
+			} else if *s.State == ec2.SnapshotStateError {
+				log.Error("snapshot failed", zap.String("id", *s.SnapshotId), zap.String("error", (*s.StateMessage)))
+				return 0, errors.Errorf("snapshot %s failed", *s.SnapshotId)
```

Comment:
Looks great Backup metadata is still generated, and contains the snapshot ids. So cleanup should delete the snapshots. https://github.com/pingcap/tidb/blob/ede35854d5b4ef165d90ac927b2ec89f7509b226/br/pkg/task/backup_ebs.go#L228-L255

Reply:
I wonder though, how much time does it take for AWS to mark the snapshots as failed.